src/utils/transactions.ts

```diff
@@ -17,7 +17,71 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { MIN_UTXO_SET_AMOUNT } from '@alephium/sdk'
+import { Transaction } from '@alephium/sdk/api/explorer'
+
+import { Address, SimpleTx } from '../contexts/addresses'
+
+type HasTimestamp = { timestamp: number }
+type TransactionVariant = Transaction | SimpleTx
+type IsTransactionVariant<T extends Transaction | SimpleTx> = T extends Transaction
+  ? Transaction
+  : T extends SimpleTx
+  ? SimpleTx
+  : never
+export type BelongingToAddress<T extends Transaction | SimpleTx> = { data: IsTransactionVariant<T>; address: Address }
 
 export const isAmountWithinRange = (amount: bigint, maxAmount: bigint): boolean => {
   return amount >= MIN_UTXO_SET_AMOUNT && amount <= maxAmount
 }
+
+export type TransactionDirection = 'out' | 'in' | 'pending'
+export type TransactionType = 'consolidation' | 'transfer' | 'sweep'
+export type TransactionStatus = 'pending' | 'confirmed'
+
+export function getTransactionsForAddresses(
+  txStatus: TransactionStatus,
+  addresses: Address[]
+): BelongingToAddress<TransactionVariant>[] {
+  return addresses
```

Comment:
nit: no need for return if converted to arrow function (which we tend to prefer in our code base)